jib-core/src/main/java/com/google/cloud/tools/jib/image/json/ManifestListGenerator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.image.json;
+
+import com.google.api.client.util.Preconditions;
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
+import com.google.cloud.tools.jib.hash.Digests;
+import com.google.cloud.tools.jib.image.Image;
+import com.google.cloud.tools.jib.image.json.V22ManifestListTemplate.ManifestDescriptorTemplate;
+import java.io.IOException;
+import java.util.List;
+
+/** Generates a manifest list for {@link Image}s. */
+public class ManifestListGenerator {
+
+  private final List<Image> images;
+
+  public ManifestListGenerator(List<Image> images) {
+    this.images = images;
+  }
+
+  /**
+   * Generates a manifest list JSON for the given {@link Image}s.
+   *
+   * @param <T> child type of {@link BuildableManifestTemplate}
+   * @param manifestTemplateClass the JSON template to translate the image to
+   * @return a manifest list JSON
+   * @throws IOException if generating a manifest list fails due to an I/O error when computing
+   *     digests
+   */
+  public <T extends BuildableManifestTemplate> ManifestTemplate getManifestListTemplate(
+      Class<T> manifestTemplateClass) throws IOException {
+    Preconditions.checkArgument(
+        manifestTemplateClass == V22ManifestTemplate.class,
+        "Build an OCI image index is not yet supported");
```

Comment:
Oh, we should consider the boundary condition when `images` is empty. I think it makes sense to just fail

```
Preconditions.checkState(!images.isEmpty(), "no images given");
```

and add a test for this. (It's important to test boundary conditions.) It also makes sense to add the same test in `BuildManifestListOrSingleManifestStepTest` to see if the exception is properly propagated to that level too.

Reply:
That makes sense , I was thinking about these cases from the `StepsRunner` standpoint. Since steps runner always ensures the `!images.isEmpty()`  we should be good but I am now realizing that these libraries can be called from anywhere in Jib so we actually need to check the parameters also inside the libraries themselves.

Reply:
Exactly. The class is independent of `StepsRunner` (or anything else unrelated). And as soon as we have the class, other parts of the Jib repo will pick it up and start using it. The implementation should be self-sufficient and error-tolerant.